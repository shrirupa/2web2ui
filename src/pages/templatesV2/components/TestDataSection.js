import React, { useEffect } from 'react';
import useEditorContext from '../hooks/useEditorContext';
import Editor from './Editor';

const TestDataSection = () => {
  const { setTestData, syncTestDataInStorage, testData } = useEditorContext();

  useEffect(() => //sync with storage during unmount
    syncTestDataInStorage
  , [syncTestDataInStorage]);

  return (
    <Editor
      mode="json"
      name="test-data-editor"
      onChange={(val) => {
        try {
          setTestData(JSON.parse(val));
        } catch (err) {} /* eslint no-empty: ["error", { "allowEmptyCatch": true }] */
      }}
      value={testData}
    />
  );
};

export default TestDataSection;

