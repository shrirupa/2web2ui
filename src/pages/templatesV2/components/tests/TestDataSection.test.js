import React from 'react';
import { shallow } from 'enzyme';
import useEditorContext from '../../hooks/useEditorContext';
import TestDataSection from '../TestDataSection';

jest.mock('../../hooks/useEditorContext');

describe('TestDataSection', () => {
  const subject = ({ stateOverride } = {}) => {
    useEditorContext.mockReturnValue({
      testData: JSON.stringify({
        subject: 'Example Subject'
      }),
      setTestData: jest.fn(),
      ...stateOverride
    });

    return shallow(<TestDataSection/>);
  };

  it('renders test data editor', () => {
    expect(subject()).toMatchSnapshot();
  });
});
