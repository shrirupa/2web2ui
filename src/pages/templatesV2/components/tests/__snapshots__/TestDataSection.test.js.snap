// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestDataSection renders test data editor 1`] = `
<Editor
  mode="json"
  name="test-data-editor"
  onChange={[Function]}
  value="{\\"subject\\":\\"Example Subject\\"}"
/>
`;
